src/net/KM_Networking.pas

```diff
@@ -1684,7 +1720,8 @@ procedure TKMNetworking.PacketRecieve(aNetClient: TKMNetClient; aSenderIndex: In
                   //We are now the host
                   fNetPlayerKind := lpk_Host;
                   fMyIndex := fNetPlayers.NiknameToLocal(fMyNikname);
-                  if Assigned(fOnReassignedHost) then fOnReassignedHost(Self); //Lobby/game might need to know that we are now hosting
+                  if Assigned(fOnReassignedHost) then
+                    fOnReassignedHost(fNetPlayers[fHostIndex].HandIndex); //Lobby/game might need to know that we are now hosting
```

Comment:
FYI we can't assume all players have a HandIndex, since spectators can be assigned to host. But since you are reverting the changes to fOnReassignedHost this isn't a problem (see my other comment)